File: ruby/testcode/ssrf_034.rb
require_relative 'shared'
require 'net/http'
require 'uri'

ALLOWED_HOSTS = %w[api.partner.com cdn.example.com].freeze

# vuln-code-snippet start ruby_ssrf_allowlist_strict
def fetch_allowlisted(req)
  uri = URI.parse(req.param('url'))
  raise 'blocked' unless ALLOWED_HOSTS.include?(uri.host) # vuln-code-snippet safe-line ruby_ssrf_allowlist_strict
  Net::HTTP.get(uri)
  BenchmarkResponse.json({ ok: true })
end
# vuln-code-snippet end ruby_ssrf_allowlist_strict

File: php/testcode/ssrf_034.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_ssrf_no_follow_location
function ssrf034(BenchmarkRequest $req): BenchmarkResponse {
    $ch = curl_init($req->param('url'));
    curl_setopt($ch, CURLOPT_FOLLOWLOCATION, false); // vuln-code-snippet safe-line php_ssrf_no_follow_location
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    $result = curl_exec($ch);
    curl_close($ch);
    return BenchmarkResponse::ok($result);
}
// vuln-code-snippet end php_ssrf_no_follow_location
